File: RemoveElement.java
/*
Runtime: 0 ms, faster than 100.00% of Java online submissions for Remove Element.
Memory Usage: 40.6 MB, less than 96.83% of Java online submissions for Remove Element.
*/
// TC: O(N) SC: O(1)
class Solution {
    public int removeElement(int[] nums, int val) {
        
        int index = 0;
        
        for(int i=0; i<nums.length; i++){
            if(nums[i] != val){
                int temp = nums[i];
                nums[i] = nums[index];
                nums[index] = temp;
                index++;
            }
        }
        
        return index;
        
    }
}

File: RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let index = 0;
    nums.forEach((el, i) => {
        if(el !== val) {
            const temp = nums[index];
            nums[index] = el;
            nums[i] = temp;
            index++;
        }
    });
    return index;
};


// Beats 94.34%of users with JavaScript
const removeElement = (nums, val) => {
    for(let i=0; i<nums.length; i++) {
        if(nums[i] === val) {
            nums.splice(i, 1);
            i--;
        }
    }
    return nums.length;
}


